src/qutip_cupy/__init__.py

```diff
@@ -5,6 +5,14 @@
 # to an official optional dependency
 import warnings
 
+try:
+    __import__("cupy")
```

Comment:
Question: Why not just `import cupy`?

Reply:
Oh ok I missed https://github.com/qutip/qutip-cupy/pull/17#discussion_r664581724. Seems hacky to me 😅